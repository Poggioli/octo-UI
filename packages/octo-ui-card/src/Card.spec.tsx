import { render } from "@testing-library/react";
import {
  Card,
  CardContainerAction,
  CardDescription,
  CardSubtitle,
  CardTitle
} from "./Card";

describe("<Card />", () => {
  it(`GIVEN a <Card />
      WHEN render is called
      THEN should render a <Card />`, () => {
    const rendered = render(
      <Card>
        <CardTitle>CardTitle</CardTitle>
        <CardSubtitle>CardSubtitle</CardSubtitle>
        <CardDescription>Card Description</CardDescription>
        <CardContainerAction>
          <button>CardContainerAction Button</button>
        </CardContainerAction>
      </Card>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Card />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Card.toString()).toBe(".octo-UI-c-gqwkJN");
  });

  it(`GIVEN a <CardContainerAction />
      WHEN toString is called
      THEN return a string`, () => {
    expect(CardContainerAction.toString()).toBe(".octo-UI-c-gqwkJN");
  });

  it(`GIVEN a <CardDescription />
      WHEN toString is called
      THEN return a string`, () => {
    expect(CardDescription.toString()).toBe(".octo-UI-c-gWerhV");
  });
  
  it(`GIVEN a <CardSubtitle />
      WHEN toString is called
      THEN return a string`, () => {
    expect(CardSubtitle.toString()).toBe(".octo-UI-c-gWerhV");
  });

  it(`GIVEN a <CardTitle />
      WHEN toString is called
      THEN return a string`, () => {
    expect(CardTitle.toString()).toBe(".octo-UI-c-gWerhV");
  });
});
