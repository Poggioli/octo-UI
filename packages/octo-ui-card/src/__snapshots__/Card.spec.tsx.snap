// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> GIVEN a <Card />
      WHEN render is called
      THEN should render a <Card /> 1`] = `
<div>
  <div
    class="octo-UI-c-gqwkJN octo-UI-c-hXwDLj octo-UI-c-gqwkJN-iTKOFX-flexDirection-column octo-UI-c-gqwkJN-eJbcS-flexWrap-initial octo-UI-c-gqwkJN-fVplCG-justifyContent-initial octo-UI-c-gqwkJN-ebbZJG-alignItems-initial octo-UI-c-gqwkJN-dYGMXV-alignContent-initial octo-UI-c-gqwkJN-bexZoo-alignSelf-initial octo-UI-c-hXwDLj-crwvoE-shaded-true octo-UI-c-hXwDLj-gwqsjS-bordered-true"
  >
    <h3
      class="octo-UI-c-gWerhV octo-UI-c-gWerhV-cMAlre-size-6 octo-UI-c-gWerhV-ikjbMWW-css"
    >
      CardTitle
    </h3>
    <h4
      class="octo-UI-c-gWerhV octo-UI-c-gWerhV-EDrvg-size-1 octo-UI-c-gWerhV-ikbIwwR-css"
    >
      CardSubtitle
    </h4>
    <p
      class="octo-UI-c-gWerhV octo-UI-c-gWerhV-cBLpOj-size-2 octo-UI-c-gWerhV-ikbIwwR-css"
    >
      Card Description
    </p>
    <div
      class="octo-UI-c-gqwkJN octo-UI-c-gqwkJN-djZMnX-flexDirection-initial octo-UI-c-gqwkJN-XefLA-flexWrap-wrap octo-UI-c-gqwkJN-awKDG-justifyContent-flexStart octo-UI-c-gqwkJN-ebbZJG-alignItems-initial octo-UI-c-gqwkJN-dYGMXV-alignContent-initial octo-UI-c-gqwkJN-bexZoo-alignSelf-initial"
    >
      <button>
        CardContainerAction Button
      </button>
    </div>
  </div>
</div>
`;
